### DISCUSSION TITLE
Is it possible to choose a selection by typing any letter?

### DISCUSSION THREAD

**Discussion by lanigb** [neoclide/coc.nvim#3946] [Q&A]
The default configuration of coc.vim makes you navigate into the completion and then insert the chosen completion by typing enter.
Is it possible to configure it so that when a completion is chosen, typing any character would insert this completion and the type  character inserted at the current insertion point. 
The goal is to economize one keystroke (enter) for each completion.

**Comment by timsu92** [comment]
Did you mean to insert the chosen completion and then the character you typed?
If so, you can use `TAB` and `shift` `tab` to select the next one and the previous one. Meanwhile, it shows the preview, so when you type the next character, it appends.

**Comment by lanigb** [comment]
Thanks for your answer.
On my computer, TAB and SHIFT TAB dont' work the way you described. I need to use the down and up arrow, and then when a completion is chosen, typing the next character  clears the selection.
Are they some settings to get the situation you described?

**Reply by timsu92** [reply]
If you meant `tab`, `shift` `tab`, I use the one in [README](https://github.com/neoclide/coc.nvim/blob/43744ba82bd348ea971b87483a13f06f3e46c94b/README.md?plain=1#L146) aside from things regarding to [coc-snippets](https://github.com/neoclide/coc-snippets).
However, I have to mention that I work on C++ most of the time. Diffrent LSPs may have different behaviors.
Maybe... you can try changing the mappings described above from `<tab>` to `<down>` and `<s-tab>` to `<up>`

**Reply by lanigb** [reply]
Thanks a lot
`I use the one in [README](https://github.com/neoclide/coc.nvim/blob/43744ba82bd348ea971b87483a13f06f3e46c94b/README.md?plain=1#L146)`
That is the solution, thanks for the link. 
I copy the part of the read me that solved my problem in case it could help somebody.


```
" Use tab for trigger completion with characters ahead and navigate.
" NOTE: Use command ':verbose imap <tab>' to make sure tab is not mapped by
" other plugin before putting this into your config.
inoremap <silent><expr> <TAB>
      \ pumvisible() ? "\<C-n>" :
      \ CheckBackspace() ? "\<TAB>" :
      \ coc#refresh()
inoremap <expr><S-TAB> pumvisible() ? "\<C-p>" : "\<C-h>"
function! CheckBackspace() abort
  let col = col('.') - 1
  return !col || getline('.')[col - 1]  =~# '\s'
endfunction
" Use <c-space> to trigger completion.
if has('nvim')
  inoremap <silent><expr> <c-space> coc#refresh()
else
  inoremap <silent><expr> <c-@> coc#refresh()
endif
" Make <CR> auto-select the first completion item and notify coc.nvim to
" format on enter, <cr> could be remapped by other vim plugin
inoremap <silent><expr> <cr> pumvisible() ? coc#_select_confirm()
                              \: "\<C-g>u\<CR>\<c-r>=coc#on_enter()\<CR>"
```
